Patch-Source: https://github.com/php/php-src/commit/aff99f5c1f52d01730eb703d920378e394690def

We need this patch to fix:

[sapi/cli/tests/017.phpt]
========DIFF========
--
     
     --------------
     Snippet no. 5:
041+ PHP Parse error:  Unmatched ')' in php shell code on line 1
     Interactive shell
     
     
044- Parse error: Unmatched ')' in php shell code on line 1
     
046- 

---
commit aff99f5c1f52d01730eb703d920378e394690def
Author: Christoph M. Becker <cmbecker69@gmx.de>
Date:   Fri Sep 2 18:30:01 2022 +0200

    Support sapi/cli/tests/017.phpt on Windows, too
    
    `escapeshellarg()` is pretty useless on Windows, and there is no way to
    support multiple lines (i.e. line breaks).  Thus, we use `proc_open()`
    instead of `shell_exec()`.
    
    We also remove some apparently superfluous empty lines from the test
    expectation; that seems to match libedit behavior on Linux.
    
    Closes GH-9474.

diff --git a/sapi/cli/tests/017.phpt b/sapi/cli/tests/017.phpt
index 8338f26e5d..d2d37b1858 100644
--- a/sapi/cli/tests/017.phpt
+++ b/sapi/cli/tests/017.phpt
@@ -8,13 +8,12 @@
 if (readline_info('done') !== NULL) {
     die ("skip need readline support using libedit");
 }
-if(substr(PHP_OS, 0, 3) == 'WIN' ) {
-    die('skip not for Windows');
-}
 ?>
 --FILE--
 <?php
 $php = getenv('TEST_PHP_EXECUTABLE');
+$ini = getenv('TEST_PHP_EXTRA_ARGS');
+$descriptorspec = [['pipe', 'r'], STDOUT, STDERR];
 
 $codes = array();
 
@@ -55,8 +54,10 @@ function a_function_with_some_name() {
 
 foreach ($codes as $key => $code) {
     echo "\n--------------\nSnippet no. $key:\n--------------\n";
-    $code = escapeshellarg($code);
-    echo `echo $code | "$php" -a`, "\n";
+    $proc = proc_open("$php $ini -a", $descriptorspec, $pipes);
+    fwrite($pipes[0], $code);
+    fclose($pipes[0]);
+    proc_close($proc);
 }
 
 echo "\nDone\n";
@@ -69,7 +70,6 @@ function a_function_with_some_name() {
 
 Hello world
 
-
 --------------
 Snippet no. 2:
 --------------
@@ -79,7 +79,6 @@ function a_function_with_some_name() {
 single
 quote
 
-
 --------------
 Snippet no. 3:
 --------------
@@ -90,7 +89,6 @@ function a_function_with_some_name() {
 the
 doc
 
-
 --------------
 Snippet no. 4:
 --------------
@@ -98,7 +96,6 @@ function a_function_with_some_name() {
 
 Done
 
-
 --------------
 Snippet no. 5:
 --------------
@@ -107,5 +104,4 @@ function a_function_with_some_name() {
 
 Parse error: Unmatched ')' in php shell code on line 1
 
-
 Done
